src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/_params.py

```diff
@@ -0,0 +1,119 @@
+
+    with self.argument_context('eventhubs namespace authorization-rule keys renew') as c:
+        c.argument('key_type', arg_type=key_arg_type)
+
+
+# region - Eventhub Create
+    with self.argument_context('eventhubs eventhub') as c:
+        c.argument('event_hub_name', arg_type=name_type, id_part='child_name_1', help='Name of Eventhub')
+
+    for scope in ['eventhubs eventhub create', 'eventhubs eventhub update']:
+        with self.argument_context(scope) as c:
+            c.argument('message_retention_in_days', type=int, help='Number of days to retain the events for this Event Hub, value should be 1 to 7 days')
+            c.argument('partition_count', type=int, help='Number of partitions created for the Event Hub, allowed values are from 1 to 32 partitions.')
+            c.argument('status', arg_type=get_enum_type(EntityStatus), help='Status of Eventhub')
+            c.argument('enabled', arg_type=get_three_state_flag(), help='A value that indicates whether capture description is enabled.')
+            c.argument('encoding', arg_type=get_enum_type(['Avro']), help='Enumerates the possible values for the encoding format of capture description.')
+            c.argument('capture_interval_seconds', type=int, help='The time window allows you to set the frequency with which the capture to Azure Blobs will happen, value should between 60 to 900 seconds')
+            c.argument('capture_size_limit_bytes', type=int, help='The size window defines the amount of data built up in your Event Hub before an capture operation, value should be between 10485760 to 524288000 bytes')
+            c.argument('destination_name', help='Name for capture destination')
+            c.argument('storage_account_resource_id', help='Resource id of the storage account to be used to create the blobs')
+            c.argument('blob_container', help='Blob container Name')
+            c.argument('archive_name_format', help='Blob naming convention for archive, e.g. {Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}. Here all the parameters (Namespace,EventHub .. etc) are mandatory irrespective of order')
+
+    # region EventHub Authorizationrule
+    for scope in ['eventhubs eventhub authorization-rule', 'eventhubs eventhub authorization-rule keys list', 'eventhubs eventhub authorization-rule keys renew']:
+        with self.argument_context(scope) as c:
+            c.argument('authorization_rule_name', arg_type=name_type, id_part='child_name_2', help='name of the EventHub AuthorizationRule')
+            c.argument('event_hub_name', id_part='child_name_1', arg_type=event_hub_name_arg_type)
+
+    with self.argument_context('eventhubs eventhub authorization-rule create') as c:
+        c.argument('rights', arg_type=rights_arg_type)
+
+    with self.argument_context('eventhubs eventhub authorization-rule keys renew') as c:
+        c.argument('key_type', arg_type=key_arg_type)
```

Comment:
Don't understand the help text for `--key` at all. It should not call out "Optional" because that's understood by the fact that it's not called out as "[Required]".

It looks like you are using the `storage account keys` commands as reference. For consistency with that, `--key-name` would be `--key` and what you currently have as `--key` would be something else...but I'm not sure what because I don't understand what it does.

Reply:
Resolved, changed to --key-name